instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -83,9 +91,16 @@ def _otel_client_attributes
 
             attributes[SemanticConventions::Trace::DB_NAME] = _otel_database_name
             attributes[SemanticConventions::Trace::PEER_SERVICE] = config[:peer_service]
+
             attributes
           end
 
+          def collection_name(sql)
+            Regexp.last_match[1] if sql =~ TABLE_NAME
```

Comment:
I could probably look this up myself but...

How does `Regexp.last_match` work? 

Is this thread safe? 

What happens if you are in a multi-threaded worker and an unrelated Regexp runs. Will it return the `last_match` from an unexpected regular expression?

Reply:
`Regexp.last_match(1)` is returning the first matched result and stopping. Since we just need the first match, this approach is faster than the previously used `.scan`, which was capturing every single match.

Re thread safety: I think so. Here's a [StackOverflow](https://stackoverflow.com/questions/9004245/is-regexp-last-match-thread-safe) answer that talks a bit about it. Because its being called inside an instance method, `Regexp.last_match()` should only be yielding the last capture that was evaluated inside the method, so no clobbering should be taking place.